api/migrations/20240711125727-delete_empty_classe.js

```diff
@@ -0,0 +1,44 @@
+const ClasseModel = require("../src/models/cle/classe");
+const YoungModel = require("../src/models/young");
+
+module.exports = {
+  async up(db, client) {
+    const emptyClasses = await ClasseModel.find({ schoolYear: "2023-2024", seatsTaken: 0 });
```

Comment:
a mon avis il ne faudrait même pas regarder le champs `seatsTaken` (car calculé et peut-être faux) et directement faire une aggregation pour récupérer les classes sans jeunes. ChatGPT et très for pour te pondre la requete :
```js
[
  {
    $lookup: {
      from: "youngs",
      localField: "_id",
      foreignField: "classeId",
      as: "youngs"
    }
  },
  {
    $match: {
      "youngs": { $size: 0 }
    }
  },
  {
    $project: {
      _id: 1
    }
  }
]
```

Reply:
J'ai essayé avec des agregation mais j'ai l'impression que les migrations le gere assez mal. deja avec la mig e statusPhase1 j'avais essayé d'en faire mais ca n'a jamais marché meme avec toute l'aide de chat. Par exemple avec ta query et toute celle que j'ai essayé je trouve 0 classe vide a chaque fois.

avec ca je supprime deja 176 classes, c'est cohérent et je suis sur de pas supprimer une classe avec des eleves, et si il reste quelques classe vide de l'année derniere c'est pas un drame